app/controllers/surveys_controller.rb

```diff
@@ -33,13 +33,22 @@ def reply
       end
 
       user_survey_submission = @survey.survey_submissions.find_by(user: current_user)
-      if user_survey_submission
-        user_survey_submission.update_attributes(updated_at: Time.current)
-      else
-        @survey.survey_submissions.create!(user: current_user)
+      begin
+        if user_survey_submission
+          user_survey_submission.update_attributes(updated_at: Time.current)
+        else
+          @survey.survey_submissions.create!(user: current_user)
+        end
+      rescue => error
```

Comment:
Hm, how can this happen?

Reply:
If we cannot create the submission record (when `create!` raises an exception)

Reply:
Sure but I don't see any condition when this will happen? Only when the user already has submitted. Isn't all of this just some for of  [`find_or_create_by`](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html#method-i-find_or_create_by-21)?